lms/djangoapps/discussion/views.py

```diff
@@ -760,6 +773,17 @@ def render_to_fragment(
             fragment = Fragment(html)
             self.add_fragment_resource_urls(fragment)
             return fragment
+        except DiscussionHiddenFromUserException:
+            log.warning(
+                'User with id={user_id} tried to view a private discussion'.format(user_id=request.user.id)
+            )
```

Comment:
Also including the `discussion_id` would make the message an even more useful diagnostic.